src/tasks/admin/delete-user.js

```diff
@@ -0,0 +1,53 @@
+const {success, failure} = require('../../lib/result');
+const taskError = require('../../lib/error/task');
+const createTaskError = require('../../lib/error/task');
+
+/**
+ * Init delete user task.
+ * @param {RealmRepository} realmRepository Realm repository
+ * @param {UserRepository} userRepository The user repository
+ * @param {AuthRepository} authRepository The auth repository
+ * @param {RealtimeConnectionRepository} realtimeConnectionRepository The realtime connection repository
+ * @param {SubscriptionRepository} subscriptionRepository The subscription repository
+ * @returns {ClientTasks#deleteUser} Task
+ */
+module.exports = ({
+  realmRepository,
+  userRepository,
+  authRepository,
+  subscriptionRepository,
+  realtimeConnectionRepository,
+}) =>
+  /**
+   * Delete a user and its auth tokens and subscriptions.
+   * @function AdminTasks#deleteUser
+   * @param {string} realmId Realm id
+   * @param {string} id User id
+   * @returns {Result<UserModel>}
+   */
+  async ({realmId, id}) => {
+    const realm = await realmRepository.findOne({id: realmId});
+    if (!realm) {
+      return failure(createTaskError({
+        code: 'UnknownRealm',
+        message: 'Cannot lookup the given realm.',
+      }));
+    }
+
+    const user = await userRepository.findOne({realmId, id});
+    if (!user) {
+      return failure(taskError({
+        code: 'UnknownUser',
+        message: 'User does not exists.',
+      }));
+    }
+
+    await Promise.all([
+      userRepository.deleteOne({realmId, id}),
+      authRepository.deleteAllByUserId({realmId, userId: id}),
+      subscriptionRepository.deleteAllByUserId({realmId, userId: id}),
+      realtimeConnectionRepository.deleteAllByUserId({realmId, userId: id}),
+    ]);
```

Comment:
**thought:** Similar to above :point_up:  this does not closes existing broker connections. Connected clients won't be kicked out and still be able to receive messages on all channels they previously subscribed.

Reply:
VerneMQ allows to [kick users via command](https://docs.vernemq.com/live-administration/managing-sessions#managing-sessions). Something similar should be possible to be implemented as a broker side API endpoint or some kind message handler within `$SYS`.